### ISSUE TITLE
Jupyterlab inspector and inspector-extension not getting installed, and no proper documentation

### ISSUE THREAD

**Issue by Koushik95** [jupyterlab/jupyterlab#4520]
https://github.com/jupyterlab/jupyterlab/tree/master/packages/inspector

For the above package, there is no proper documentation as to how to install it and how it will load in the jupyterlab environment once it is installed. Please if someone could update the above request of mine, it would be helpful. Samething goes for the inspector-extension as well :- https://github.com/jupyterlab/jupyterlab/tree/master/packages/inspector-extension

**Comment by jasongrout** [comment]
The inspector is installed and enabled by default. You're right that we don't have a description of what it is in the user documentation, so I think that would be a good goal for this issue.

Open the inspector by running the "Open Inspector" command from the command palette. Basically, it watches what you are typing in a notebook and gives you help on the current function, etc.

**Comment by jasongrout** [comment]
Here is a short video of using it: https://github.com/jupyterlab/jupyterlab/pull/537#issuecomment-241463288

**Comment by dhirschfeld** [comment]
I think @Koushik95 is getting confused by the description:
> *A JupyterLab package which provides a variable inspector*

...and what they're really after is a variable *explorer* a la #443.

I think the terminology is a bit misleading myself and would instead call the current "Variable Inspector" more of a "Help Browser"

**Comment by jasongrout** [comment]
Ah, the description has been edited quite a bit now to make it much clearer that a variable explorer was what was desired.

The current extension is "Inspector", not "Variable Inspector" IIRC. I agree that even "Inspector" has confused more than one person.

**Comment by mimischi** [comment]
Toggling the inspector opens a new empty tab for me, instead of providing a vertical tab on the right (as shown here: https://github.com/jupyterlab/jupyterlab/pull/537#issuecomment-241463288). Is that the expected behavior currently?

**Comment by Koushik95** [comment]
@jasongrout , yes what I was looking for is a variable explorer. I'm well aware of the default inspector tab provided by jupyterlabs and was not referring to that. I was strictly speaking in terms of a tab which tracks the properties of variables defined in the notebooks created in the jupyterlabs environment. 
Similar to the varInspector extension provided in the nbextensions section for Jupyter notebooks :- 
https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/varInspector

**Comment by liffiton** [comment]
I think this is covered in issue #443, then, fwiw.

**Comment by goerz** [comment]
When I start the "Inspector", I only get a white page. Is this currently expected to work?

**Comment by saulshanabrook** [comment]
@goerz Yes. It should fill in when you type a function that has documentation.

**Comment by saulshanabrook** [comment]
@jasongrout did you want to update some documentation (you tagged it as such) or should this be closed as a duplicate?

**Comment by jasongrout** [comment]
I still think it would be good to have some docs on it we can point people to. Perhaps the is a good first issue?

**Comment by ahamedmoosa** [comment]
Check this out @Koushik95  
https://github.com/lckr/jupyterlab-variableInspector
It works pretty cool, I guess you would have found it already

**Comment by aromaticSesame** [comment]
Thanks for opening the issue. I met the same problem and would really hope this function be added officially.
